packages/core/useSort/index.ts

```diff
@@ -0,0 +1,63 @@
+import { computed } from '@vue/reactivity'
+import type { MaybeRef } from '@vueuse/shared'
+import { isRef, unref, watchEffect } from 'vue-demi'
+
+export type UseSortCompareFn<T> = (a: T, b: T) => number
+
+export type UseSortFn<T> = (arr: T[], compareFn: UseSortCompareFn<T>) => T[]
+
+export interface UseSortOptions<T> {
+  /**
+   * compare function
+   */
+  compareFn?: UseSortCompareFn<T>
+  /**
+   * change the value of the source array
+   * @default false
+   */
+  dirty?: boolean
+}
+
+const defaultCompare: UseSortCompareFn<number> = (a, b) => a - b
+
+/**
+ * reactive sort array
+ *
+ * @see https://vueuse.org/useSort
+ * @param source source array
+ * @param sortFn sort function
+ * @param options
+ */
+export function useSort<T>(source: MaybeRef<T[]>, sortFn: UseSortFn<T>, options: UseSortOptions<T> = {}) {
```

Comment:
```suggestion
export function useSorted<T>(source: MaybeRef<T[]>, sortFn: UseSortFn<T>, options: UseSortOptions<T> = {}) {
```

`sort` sounds like mutating, while we are trying to be immutable. `sorted` sounds like a better fit https://github.com/tc39/proposal-change-array-by-copy

Reply:
I would also like to see it matches with JS's sort more with an optional comparator `(a, b) => number`

Reply:
> I would also like to see it matches with JS's sort more with an optional comparator `(a, b) => number`

Already have
https://github.com/vueuse/vueuse/pull/1465/files#diff-57d9b1ba0ff31d653b57c370d5eaf1e56421d821a8aa79aba512a3da23a2747dR13